### PR TITLE
Changes the bot to use discords API

### LINKED ISSUE #58: No way to login with a token
There currently doesn't appear to be a way to login to discord with a token on an API application account. :/

### PR DESCRIPTION
**Pull request by TheDoctorsLife**
Fixes #58 :-D

### CODE REVIEW

**Comment by TheDoctorsLife** [comment]
I know how to fix this, incoming pr soon.

**Comment by TheDoctorsLife** [comment]
If someone would like to help me figure out how to pass travis right now, that would be great. I'm stuck.

**Comment by ekmartin** [comment]
Hi, cool! Would it be smart to support both initially, so that people have time to convert to proper bot accounts? Maybe you could add a check in `connect()` that uses `loginWithToken` only if there's a token set, and `login` if not?

As for the tests, you need to update the stubs. Can add a loginWithToken method here: https://github.com/reactiflux/discord-irc/blob/master/test%2Fstubs%2Fdiscord-stub.js#L24, or stub stuff in `beforeEach` of the tests like it's done here: https://github.com/reactiflux/discord-irc/blob/master/test%2Fbot-events.test.js#L28

Feel free to PM me on Discord if you have more questions about the tests (ekmartin) :)

**Comment by ekmartin** [comment]
If it's easy to create and set up a bot user with Discord's new system I guess we could just release version 1.0 and break backwards compatibility with the old email/password setup too. 

Could you add a section in the README explaining how to create a bot user and invite it to the server?

**Comment by TheDoctorsLife** [comment]
Sure! Thanks for the help, I'll do those to later today once I get home.

**Comment by TheDoctorsLife** [comment]
This is ready to be merged, the error that travis and frigg is present on the current master branch.

**Comment by ekmartin** [comment]
Master looks pretty green to me (https://travis-ci.org/reactiflux/discord-irc/builds/124643299) - looks like a couple of the tests are failing with your branch: https://travis-ci.org/reactiflux/discord-irc/jobs/126644672 (check the bottom). Run `npm test` or `npm run coverage` to test it locally.

File: README.md [hunk 1]
```diff
@@ -40,8 +39,7 @@ Create a discord user and join the server you want to connect to IRC.
   {
     "nickname": "test2",
     "server": "irc.testbot.org",
-    "discordEmail": "user@mail.com",
-    "discordPassword": "letmein123",
+    "discordToken: "botwantsin123",
```
Comment (ekmartin) [inline README.md]: Missing a quote here (`"`)

**Comment by TheDoctorsLife** [comment]
Hmm, I just clicked the frigg status in the README and thought that, sorry.

**Comment by ekmartin** [comment]
No problem, that's a bit confusing indeed (it goes to the latest build, not the latest on master).

**Comment by TheDoctorsLife** [comment]
Alright looks good now, still waiting for frigg, but I think it'll pass to.

**Comment by TheDoctorsLife** [comment]
All checks have passed, this is ready to be merged now.

**Comment by ekmartin** [comment]
Thanks! Added a changelog entry and released version 1.0.0.

**Comment by Deph0** [comment]
Can we please have both?

**Comment by TheDoctorsLife** [comment]
It probably be a good idea to just force everyone to convert, since discord will begin to close the portal to non-converted bot accounts soon.

**Pull request merged**